### ISSUE TITLE
replace os.path with pathlib in all v02 scripts

### ISSUE THREAD

**Issue by jhrehanoaa** [NOAA-OWP/t-route#210]
This is an extension of #78 remove os.path.dirname with pathlib in remaining files. Three files still contain that call: augment, build tests, and nhd_network_utilities. This new issue is more general and aims to replace os.path with pathlib across all v02 scripts.

 

Anywhere we are using 
```
root = os.path.dirname(os.path.abspath(__file__))
```
 or any variant of that (i.e., `os.path.aabspath("")`), should be changed to the more robust 
```
import pathlib
root = pathlib.Path("../..").resolve()
```

... taking care to make sure that the proper relative path (e.g., `../..` vs. `../../../`) is resolved based on the script location.

`pathlib.Path(".")` will resolve to the current working directory (`os.getcwd()`), not the script location.
For the script location, `pathlib.Path(__file__).resolve()` works.

Also note that `__file__` is not defined in interactive python sessions. So these statements should be guarded with a
```python
if __name__ == "__main__":
    <...>
```
block

**Comment by jhrehanoaa** [comment]
src/python_framework_v02/troute/nhd_network_utilities_v02.py

**Comment by jameshalgren** [comment]
closed with #243
